src/components/SearchBar.jsx

```diff
@@ -16,59 +29,148 @@ export default class SearchBar extends Component {
+      type: this.state.type,
+      order: this.state.order,
+      videoDuration: this.state.videoDuration
     };
     const ya = new YoutubeApi('search', params);
     ya.call().then(result => {
-      console.log(result);
+      this.setState({videos: result.data.items});
+      // console.log(this.state.videos);
     }).catch(error => {
       console.log(error);
     });
 
     this.setState({ value: '' });
   }
 
+  //search video type
+  handleType = event => {
+    this.setState({
+      type: event.target.value
+    });
+
+    if (event.target.value !== 'video') {
+      this.setState({
+        videoDuration: 'any'
+      });
+    }
+  };
+
+  handleDuration = event => {
+    this.setState({
+      videoDuration: event.target.value
+    });
+
+    if (event.target.value !== 'any') {
+      this.setState({
+        type: 'video'
+      });
+    }
+  };
+
   render() {
     const styles = {
       container: {
-        marginTop: '150px',
+        marginTop: '150px'
       },
       form: {
         marginTop: '30px',
-        textAlign: 'center', 
+        textAlign: 'center'
       },
       buttonStyle: {
         marginTop: '10px',
-        backgroundColor: '#2196f3', 
-        color: '#fff',
-      },
+        backgroundColor: '#2196f3',
+        color: '#fff'
+      }
     };
 
     return (
       <div>
-        <Grid container justify="center" alignItems="center" style={styles.container}>
+        <Grid
+          container
+          justify="center"
+          alignItems="center"
+          spacing={24}
+          style={styles.container}
+        >
           <Grid item xs={8}>
             <Typography align="center" variant="display1">
               This is a Youtube Clone site and is under development.
             </Typography>
             <form onSubmit={this.onFormSubmit.bind(this)} style={styles.form}>
-              <Input 
-                placeholder="Search" 
-                fullWidth 
-                style={{paddingLeft: '3px'}}
-                type="search" 
+              <Input
+                placeholder="Search"
+                fullWidth
+                style={{ paddingLeft: '3px' }}
+                type="search"
                 onChange={event => this.setState({ value: event.target.value })}
                 value={this.state.value}
               />
-              <Button 
+
+              <Button
                 type="submit"
                 variant="contained"
                 style={styles.buttonStyle}
               >
-              Search
+                Search
               </Button>
             </form>
           </Grid>
+          
+          <Grid item xs={8}>
+            <div style={{ textAlign: 'center' }}>
+              {/*Sort feature*/}
+              <FormControl style={{ margin: '0 5px' }}>
+                <InputLabel htmlFor="sort-feature">Sort by</InputLabel>
+                <Select
+                  value={this.state.order}
+                  onChange={event =>
```

Comment:
it seems good that users can show video result without clicking search button again in video result page.
it might be hard to do it from now, so this is not required.

Reply:
It could be done if the entire search function doesn't require a search button, I think it wouldn't have the look and feel of YouTube search function without the button.